app/lib/rocketchat/services/connect.ts

```diff
@@ -0,0 +1,273 @@
+						const serverVersion = store.getState().server.version;
+						if (compareServerVersion(serverVersion as string, 'greaterThanOrEqualTo', '3.9.0')) {
+							const user = params.user ?? params.username;
+							const password = params.password ?? params.ldapPass ?? params.crowdPassword;
+							params = { user, password };
+						}
+
+						return resolve(loginTOTP({ ...params, code: code?.twoFactorCode }, loginEmailPassword));
+					}
+
+					return resolve(
+						loginTOTP({
+							totp: {
+								login: {
+									...params
+								},
+								code: code?.twoFactorCode
+							}
+						})
+					);
+				} catch {
+					// twoFactor was canceled
+					return reject();
+				}
+			} else {
+				reject(e);
+			}
+		}
+	});
+}
+
+function loginWithPassword({ user, password }: { user: string; password: string }) {
+	let params: ICredentials = { user, password };
+	const state = store.getState();
+
+	if (state.settings.LDAP_Enable) {
+		params = {
+			username: user,
+			ldapPass: password,
+			ldap: true,
+			ldapOptions: {}
+		};
+	} else if (state.settings.CROWD_Enable) {
+		params = {
+			username: user,
+			crowdPassword: password,
+			crowd: true
+		};
+	}
+
+	return loginTOTP(params, true);
+}
+
+async function loginOAuthOrSso(params: ICredentials, isFromWebView = true) {
+	const result = await loginTOTP(params, false, isFromWebView);
+	store.dispatch(loginRequest({ resume: result.token }, false, isFromWebView));
+}
+
+function abort() {
+	if (sdk) {
+		return sdk.abort();
+	}
+	return new AbortController();
+}
+
+function checkAndReopen() {
+	return sdk.checkAndReopen();
+}
+
+function disconnect() {
+	return sdk.disconnect();
+}
+
+async function getServerInfo(server: string) {
+	try {
+		const response = await RNFetchBlob.fetch('GET', `${server}/api/info`, { ...RocketChatSettings.customHeaders });
+		try {
+			// Try to resolve as json
+			const jsonRes = response.json();
+			if (!jsonRes?.success) {
+				return {
+					success: false,
+					message: I18n.t('Not_RC_Server', { contact: I18n.t('Contact_your_server_admin') })
+				};
+			}
+			if (compareServerVersion(jsonRes.version, 'lowerThan', MIN_ROCKETCHAT_VERSION)) {
+				return {
+					success: false,
+					message: I18n.t('Invalid_server_version', {
+						currentVersion: jsonRes.version,
+						minVersion: MIN_ROCKETCHAT_VERSION
+					})
+				};
+			}
+			return jsonRes;
+		} catch (error) {
+			// Request is successful, but response isn't a json
+		}
+	} catch (e: any) {
+		if (e?.message) {
+			if (e.message === 'Aborted') {
+				store.dispatch(selectServerFailure());
+				throw e;
+			}
+			return {
+				success: false,
+				message: e.message
+			};
+		}
+	}
+
+	return {
+		success: false,
+		message: I18n.t('Not_RC_Server', { contact: I18n.t('Contact_your_server_admin') })
+	};
+}
+
+async function getWebsocketInfo({ server }: { server: string }) {
+	const sdk = new RocketchatClient({ host: server, protocol: 'ddp', useSsl: useSsl(server) });
```

Comment:
Up there you import `sdk` and here in this function you create it with the name `sdk`, javascript can understand, but this is not a good practice.